### PR TITLE
refactor(sticky-tabs-2): Swap in artist tabs 

### PR DESCRIPTION
**Pull request by damassi**
### Description

This swaps out the old sticky tabs with the new implementation. Along the way we had to slightly refactor the artist page so that the filter show / hide state was hoisted into the ArtworkFilter store versus being placed in local setState; this was necessary due to us needing to call open / show artwork filters from a few locations. 

Also, added a new `<SubTabBar>` component which can be used to display sticky content below the tab bar, per page. Quite tricky, but straightforward in the end once I started thinking less about how `stickyTabsIndices` is broken in this library, and more about how to recreate it from scratch. 

https://github.com/artsy/eigen/assets/236943/3cd4a31e-312d-4f10-bfb6-6dc212add94c

cc @artsy/mobile-platform 

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Components/Tabs/TabsContainer.tsx [hunk 1]
```diff
@@ -8,22 +8,26 @@ import {
...
   tabScrollEnabled?: boolean
 }
 
 export const TabsContainer: React.FC<TabsContainerProps> = ({
-  renderHeader,
   children,
   initialTabName,
+  renderHeader,
   tabScrollEnabled = false,
-  renderBelowTabBar,
```
Comment (damassi) [inline src/app/Components/Tabs/TabsContainer.tsx:20]: @gkartalis - I removed this in this PR, but now that i'm thinking about it it might be more generally useful, since we might want a component to appear regardless of what tab we're on? Feel free to add this back (or not!) and maybe add a comment about SubTabBar in the interface for this, so the purposes of each are clearly distinguished.

File: src/app/utils/tests/getMockCollapsibleTabView.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+      Lazy: React.View,
+      ScrollView: React.ScrollView,
+      Tab: MockedTabs,
+    },
+    useCurrentTabScrollY: () => ({ value: 0 }),
+    useFocusedTab: () => "SomeFocusedTab",
+    useHeaderMeasurements: () => ({ height: { value: 0 }, top: { value: 0 } }),
+    useTabNameContext: () => "Tab",
+  }
+}
```
Comment (damassi) [inline src/app/utils/tests/getMockCollapsibleTabView.tsx:23]: Moved into its own file since we might need to adjust the behavior for a specific test, and don't want to have to duplicate all of this (necessary) setup

File: src/app/Components/Artist/ArtistInsights/ArtistInsights.tests.tsx [hunk 3]
```diff
@@ -8,13 +8,22 @@ import { createMockEnvironment } from "relay-test-utils"
 import { ArtistInsightsFragmentContainer } from "./ArtistInsights"
 import { ArtistInsightsAuctionResultsPaginationContainer } from "./ArtistInsightsAuctionResults"
 
+jest.mock("react-native-collapsible-tab-view", () => {
+  const getMockCollapsibleTabs =
+    require("app/utils/tests/getMockCollapsibleTabView").getMockCollapsibleTabs
+  return {
+    ...getMockCollapsibleTabs(),
+    useFocusedTab: () => "Insights",
```
Comment (damassi) [inline src/app/Components/Artist/ArtistInsights/ArtistInsights.tests.tsx:16]: Like here, for example

**Review by gkartalis — APPROVED** [review]
